cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/PrometheusMetricSamplerTest.java

```diff
@@ -0,0 +1,491 @@
+
+    private static final int FIXED_VALUE = 94;
+    private static final long START_EPOCH_SECONDS = 1603301400L;
+    private static final long START_TIME_MS = START_EPOCH_SECONDS * MILLIS_IN_SECOND;
+    private static final long END_TIME_MS = START_TIME_MS + 59 * MILLIS_IN_SECOND;
+
+    private static final int TOTAL_BROKERS = 3;
+    private static final int TOTAL_PARTITIONS = 3;
+
+    private static final String TEST_TOPIC = "test-topic";
+
+    private PrometheusMetricSampler _prometheusMetricSampler;
+    private PrometheusAdapter _prometheusAdapter;
+    private Map<RawMetricType, String> _prometheusQueryMap;
+
+    /**
+     * Set up mocks
+     */
+    @Before
+    public void setUp() {
+        _prometheusAdapter = mock(PrometheusAdapter.class);
+        _prometheusMetricSampler = new PrometheusMetricSampler();
+        _prometheusQueryMap = new DefaultPrometheusQuerySupplier().get();
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testConfigureWithPrometheusEndpointNoPortFails() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "http://kafka-cluster-1.org");
+        addCapacityConfig(config);
+        _prometheusMetricSampler.configure(config);
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testConfigureWithPrometheusEndpointNegativePortFails() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "http://kafka-cluster-1.org:-20");
+        addCapacityConfig(config);
+        _prometheusMetricSampler.configure(config);
+    }
+
+    @Test
+    public void testConfigureWithPrometheusEndpointNoSchemaDoesNotFail() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "kafka-cluster-1.org:9090");
+        addCapacityConfig(config);
+        _prometheusMetricSampler.configure(config);
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testConfigureWithNoPrometheusEndpointFails() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        addCapacityConfig(config);
+        _prometheusMetricSampler.configure(config);
+    }
+
+    @Test(expected = SamplingException.class)
+    public void testGetSamplesQueryThrowsException() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "http://kafka-cluster-1.org:9090");
+        addCapacityConfig(config);
+        _prometheusMetricSampler.configure(config);
+
+        MetricSamplerOptions metricSamplerOptions = buildMetricSamplerOptions();
+        _prometheusMetricSampler._prometheusAdapter = _prometheusAdapter;
+        expect(_prometheusAdapter.queryMetric(anyString(), anyLong(), anyLong()))
+            .andThrow(new IOException("Exception in fetching metrics"));
+
+        replay(_prometheusAdapter);
```

Comment:
Missing `verify(_prometheusAdapter)` for the replayed mock.

Reply:
Fixed.